### PR TITLE
Digitizer mch

### PR DESCRIPTION
**Pull request by mwinn2**
Hi @aphecetche , 
this is the most naive implementation.
It is very ugly in Digitizer.cxx to have this loop into the loop. Alternatively, one could give the process Hit also the MClabel COntainer and fill it inside the digit loop. This is certainly faster, but I am not sure, if this not becomes quite ugly in terms of reading. 
let me know.

### CODE REVIEW

**Comment by mwinn2** [comment]
Hi, 
Ok, I didn't apparently pulled from dev for my branch. I will check if I can clean it up and redo it...

**Comment by mwinn2** [comment]
@aphacetche: should be fine now.

File: Detectors/MUON/MCH/Simulation/src/Digitizer.cxx [hunk 1]
```diff
@@ -151,9 +158,10 @@ int Digitizer::processHit(const Hit& hit, int detID, double event_time, int labe
       q *= chargenon;
     }
     auto signal = resp.response(q);
-    digits.emplace_back(padid, signal, labelIndex);
+    digits.emplace_back(padid, signal);
+    ++ndigits;
   });
-  return mDigits.size();
+  return ndigits;
```
Comment (aphecetche) [inline Detectors/MUON/MCH/Simulation/src/Digitizer.cxx:163]: don't understand why you changed back to an extra variable to count the elements : the vector does know its size just fine, doesn't it ?

File: Detectors/MUON/MCH/Simulation/src/Digitizer.cxx [hunk 2]
```diff
@@ -88,15 +88,19 @@ void Digitizer::process(const std::vector<Hit> hits, std::vector<Digit>& digits)
     int labelIndex = mMCTruthContainer.getIndexedSize();
     //index for this hit
     int detID = hit.GetDetectorID();
-    processHit(hit, detID, mEventTime, labelIndex);
-    MCCompLabel label(hit.GetTrackID(), mEventID, mSrcID);
-    mMCTruthContainer.addElementRandomAccess(labelIndex, label);
+    int ndigits =  processHit(hit, detID, mEventTime);
+    //TODO need one label per Digit
+    // can use nDigit output of processHit
+    for ( int i=0; i<ndigits; ++i ) {
```
Comment (aphecetche) [inline Detectors/MUON/MCH/Simulation/src/Digitizer.cxx]: ok, I'm getting confused here. 
First, I would change the name of the data member mMCTruthContainer to make it clearer what it is a container for, e.g. something like mTrackLabels ?
Next, the label construction should happen outside of the loop.

**Review by aphecetche — CHANGES_REQUESTED** [review]

**Comment by aphecetche** [comment]
@mwinn2 sorry for the late reply 

btw, I did not get the direct notification as my handle is not @aphacetche but @aphecetche ;-) 
(but that's not the reason for the late reply, to be honest ;-) )

File: Detectors/MUON/MCH/Simulation/src/Digitizer.cxx (hunk 1 above)
Comment (mwinn2) [reply Detectors/MUON/MCH/Simulation/src/Digitizer.cxx:163]: Hi, 
The reason is that I want to count per hit and not the total. The vector accumulates the entries over several hits.

File: Detectors/MUON/MCH/Simulation/src/Digitizer.cxx (hunk 2 above)
Comment (mwinn2) [reply Detectors/MUON/MCH/Simulation/src/Digitizer.cxx]: Ok, I will have a look later.

File: Detectors/MUON/MCH/Simulation/src/Digitizer.cxx (hunk 1 above)
Comment (aphecetche) [reply Detectors/MUON/MCH/Simulation/src/Digitizer.cxx:163]: ah, yes, ok.

File: Detectors/MUON/MCH/Simulation/src/Digitizer.cxx (hunk 2 above)
Comment (mwinn2) [reply Detectors/MUON/MCH/Simulation/src/Digitizer.cxx]: Ok, for the generation, sorry. 
I renamed it also now. 
Is there something else, which is strange to you?

Comment (aphecetche) [reply Detectors/MUON/MCH/Simulation/src/Digitizer.cxx]: nope. looks good to me.

Comment (mwinn2) [reply Detectors/MUON/MCH/Simulation/src/Digitizer.cxx]: Ok, shall I do a PR with this already or do we wait for other developments?
About pile-up, I wrote a novel on JIRA, so might have a look and let me know what are the time scales.

Comment (aphecetche) [reply Detectors/MUON/MCH/Simulation/src/Digitizer.cxx]: You can do a PR. Not 100% sure I’ll be able to review it next week, we’ll see.
For your novel, I’ll read it after my holidays ;-)

**Comment by mwinn2** [comment]
Ok, thanks. Have a nice break.
Michael

**Pull request closed without merging**